californium-core/src/test/java/org/eclipse/californium/core/test/ClientAsynchronousTest.java

```diff
@@ -135,40 +140,51 @@ public void testAsyncPostUpdatesResource() throws Exception {
 
 	@Test
 	public void testAsyncObserveTriggersOnLoad() throws Exception {
-		final CountDownLatch latch = new CountDownLatch(1);
-		final CountDownLatch expectedNotifications = new CountDownLatch(3);
-		final AtomicInteger receivedNotifications = new AtomicInteger();
+		final CyclicBarrier barrier = new CyclicBarrier(2);
+		final AtomicInteger onLoadCounter = new AtomicInteger();
 
 		// Observe the resource
 		CoapObserveRelation obs1 = client.observe(new TestHandler("Test Observe") {
-			@Override public void onLoad(final CoapResponse response) {
-				System.err.format("onLoad(%d)%n", receivedNotifications.get());
-				if (CONTENT_1.equals(response.getResponseText())
-					&& response.advanced().getOptions().hasObserve()) {
-					if (latch.getCount() > 0) {
-						System.err.format("Response onLoad(%d)%n", receivedNotifications.get());
-						latch.countDown();
-					} else {
-						expectedNotifications.countDown();
-						receivedNotifications.incrementAndGet();
-						System.err.format("Notification onLoad(%d)%n", receivedNotifications.get());
+
+			@Override
+			public void onLoad(final CoapResponse response) {
+				String responseDescription = String.format("onLoad(%d) '%s'", onLoadCounter.incrementAndGet(),
+						response.getResponseText());
+				System.err.println(responseDescription);
+				if (response.getResponseText().startsWith(CONTENT_1) && response.advanced().getOptions().hasObserve()) {
+					try {
+						assertAwait("assert missing", barrier, 2000, TimeUnit.MILLISECONDS);
+					} catch (InterruptedException e) {
 					}
 				}
 			}
 		});
-		assertTrue(latch.await(1500, TimeUnit.MILLISECONDS));
+
+		assertAwait("response missing", barrier, 2000, TimeUnit.MILLISECONDS);
+
 		System.err.println("changed 1");
+		resource.setContent(CONTENT_1 + " - 1");
```

Comment:
That's was already like this before but why using System.err instead of System.out ?

Reply:
I don't know :-). 
If it's intended to be corrected, I would go for the logger ...
But in the very first place, I want to get rid of the race condition.